route.cc

```diff
@@ -18,161 +18,96 @@
  */
 
 #include "defs.h"
-#include "grtcirc.h"
-#include "session.h"
-#include <cstdio>
-
-queue my_route_head;
-queue my_track_head;
-static int rte_head_ct;
-static int rte_waypts;
-static int trk_head_ct;
-static int trk_waypts;
+#include "grtcirc.h"            // for RAD, gcdist, heading_true_degrees, radtometers
+#include "queue.h"              // for dequeue, queue, QUEUE_FOR_EACH, QUEUE_EMPTY, ENQUEUE_HEAD, ENQUEUE_TAIL, QUEUE_INIT, QUEUE_LAST, QUEUE_NEXT
+#include "session.h"            // for curr_session, session_t (ptr only)
+#include "src/core/datetime.h"  // for DateTime
+#include "src/core/optional.h"  // for optional, operator>, operator<
+#include <QtCore/QDateTime>     // for QDateTime
+#include <QtCore/QList>         // for QList<>::iterator
+#include <QtCore/QString>       // for QString
+#include <QtCore/QtGlobal>      // for foreach
+#include <algorithm>            // for sort
+#include <cstddef>              // for nullptr_t
+
+RouteList* global_route_list;
```

Comment:
They can't be static because they are used by the template route_disp_all which gets included all over.

But we do have a bunch of global routines, e.g. route_add_head, ..., that just do one liners on global_route_list/global_track_list.  These could be eliminated in a massive edit that exposed global_route_list/global_track_list to the world instead.  That would  open up more of the power already in RouteList, e.g. iterators.  So far these aren't used on global_route_list/global_track_list because they are semi-hidden.  I'm not sure I want to do that, but I certainly don't want to do it in this PR.